### PR TITLE
Fire ModifiedEvent when field is set to null in a PATCH request.

### LINKED ISSUE #802: ObjectModifiedEvent is not fired when field is set to null
When a field is set to null in a PATCH request, no ObjectModifiedEvent is fired (or if any other field is changed in the same request, the field that was set to null will not be in the attributes).

### PR DESCRIPTION
**Pull request by phgross**
Setting a field to null in a PATCH request, was not tracked as field change in the DX Content deserializer and therefore no ModifiedEvent was fired. Closes #802 

Besides that, I split up the call method to make code analysis checks happy.

### CODE REVIEW

**Comment by phgross** [comment]
@jenkins-plone-org please run jobs

File: src/plone/restapi/deserializer/dxcontent.py [hunk 1]
```diff
@@ -33,14 +33,46 @@ def __init__(self, context, request):
...
+            validator = queryMultiAdapter(
+                (self.context, self.request, None, schema, None), IManagerValidator
+            )
+            for error in validator.validate(field_data):
+                errors.append({"error": error, "message": str(error)})
+
+        if errors:
+            raise BadRequest(errors)
+
+        # We'll set the layout after the validation and and even if there
```
Comment (tisto) [inline src/plone/restapi/deserializer/dxcontent.py]: @phgross two "and" :)

Comment (phgross) [reply src/plone/restapi/deserializer/dxcontent.py]: 😅 updated ...

File: src/plone/restapi/deserializer/dxcontent.py [hunk 2]
```diff
@@ -123,33 +147,15 @@ def __call__(
...
-                (self.context, self.request, None, schema, None), IManagerValidator
-            )
-            for error in validator.validate(field_data):
-                errors.append({"error": error, "message": str(error)})
+        return schema_data, errors
 
-        if errors:
-            raise BadRequest(errors)
+    def mark_field_as_changed(self, schema, fieldname):
+        """Collect the names of the modified fields use prefixed name because
```
Comment (lukasgraf) [inline src/plone/restapi/deserializer/dxcontent.py]: `...fields use prefixed...` ->
fields.
Use prefixed...

**Comment by tisto** [comment]
@jenkins-plone-org please run jobs

**Comment by tisto** [comment]
@phgross do you want me to do a release right after the merge?

**Comment by phgross** [comment]
@tisto yes, please, thank you!

**Comment by tisto** [comment]
@phgross done: https://pypi.org/project/plone.restapi/4.5.1/#id1

**Pull request merged**